cli/kubectl_kadalu/storage_add.py

```diff
@@ -0,0 +1,122 @@
+import os
+import yaml
+import subprocess
+import tempfile
+import sys
+
+
+KUBECTL_CMD = "kubectl"
+
+
+def storage_add_args(subparsers):
+    parser_add_storage = subparsers.add_parser('storage-add')
+    parser_add_storage.add_argument(
+        "name",
+        help="Storage Name"
+    )
+    parser_add_storage.add_argument(
+        "--type",
+        help="Storage Type",
+        choices=["Replica1", "Replica3"],
+        default="Replica1"
+    )
+    parser_add_storage.add_argument(
+        "--device",
+        help=("Storage device in <node>:<device> format, "
+              "Example: --device kube1.example.com:/dev/vdc"),
+        default=[],
+        action="append"
+    )
+    parser_add_storage.add_argument(
+        "--path",
+        help=("Storage path in <node>:<path> format, "
+              "Example: --path kube1.example.com:/exports/data"),
+        default=[],
+        action="append"
+    )
+    parser_add_storage.add_argument(
+        "--pvc",
+        help="Storage from pvc, Example: --pvc local-pvc-1",
+        default=[],
+        action="append"
+    )
+
+
+def storage_add_data(args):
+    content = {
+        "apiVersion": "kadalu-operator.storage/v1alpha1",
+        "kind": "KadaluStorage",
+        "metadata": {
+            "name": args.name
+        },
+        "spec": {
+            "type": args.type,
+            "storage": []
+        }
+    }
+
+    # Device details are specified
+    if args.device:
+        for devdata in args.device:
+            node, dev = devdata.split(":")
+            content["spec"]["storage"].append(
+                {
+                    "node": node,
+                    "device": dev
+                }
+            )
+        return content
+
+    # Path is spacified instead of Raw device
+    if args.path:
+        for pathdata in args.path:
+            node, path = pathdata.split(":")
+            content["spec"]["storage"].append(
+                {
+                    "node": node,
+                    "path": path
+                }
+            )
+        return content
+
+    # If PVC is specified instead of Raw device and Path
+    if args.pvc:
+        for pvcdata in args.pvc:
+            node = ""
+            pvc = pvcdata
+            if ":" in pvcdata:
+                node, pvc = pvcdata.split(":")
+
+            content["spec"]["storage"].append(
+                {
+                    "node": node,
```

Comment:
no need to provide 'node' option for PVC. ideally that gets ignored. Because what happens in k8s is, where the PVC is provisioned, thats where the pod should start.

Reply:
ack. I will remove this. Added since an example contains node https://github.com/kadalu/kadalu/blob/master/examples/sample-storage-file-device.yaml#L25

Reply:
Yes, realized it... I should fix that and run tests to validate my claim too.

Reply:
OK, seems like we have some issues when only PVC option is given. Needs more test. When I have both node and PVC, it works fine.

Let this patch consider not providing node when PVC is given. Will fix the PVC example issue separately.